client/core/bookie.go

```diff
@@ -408,6 +409,24 @@ func handleUnbookOrderMsg(_ *Core, dc *dexConnection, msg *msgjson.Message) erro
 		Payload:  &MiniOrder{Token: token(note.OrderID)},
 	})
 
+	// Mark the unbooked order as revoked if not executed, canceled
+	// or already revoked.
+	// NOTE: UnbookOrder notes are sent before match requests, so
+	// unbooked orders that are unbooked because they've just been
+	// executed or canceled may NOT YET have their statuses updated
+	// to Executed or Canceled. Such orders will be revoked here, but
+	// will be marked as Executed/Canceled when the tracker receives
+	// the last set of matches in tracker.negotiate(). If the last set
+	// of matches are received before this UnbookOrder note, then the
+	// order would not be revoked because it would already have been
+	// marked as Executed or Canceled.
+	var oid order.OrderID
+	copy(oid[:], note.OrderID)
+	tracker, _, _ := dc.findOrder(oid)
+	if tracker != nil && tracker.revoke() {
```

Comment:
`unbook_order` is an `orderbook` (subscription) notification, and it should not be necessary to be subscribed to a market's `orderbook` feed for core's trade/match negotiation to function, only being auth'd with the server.

Even if that weren't the case, I'm not terribly comfortable with the ambiguities here and and switching of all unbooked orders through revoked status.  Let's try for a solution that doesn't set status as revoked for all unbooked orders.

Reply:
I agree this is a hack, and seeing/remembering that a client is not required to subscribe to a market's orderbook even if trading in that market makes this hack insufficient at best.

I'm gonna revert/undo that change set; and we can get back to discussing how to determine revoked orders on the client so coins are unlocked.